### PR TITLE
CWL: Reformat w/ Black and better support for tmpdir_prefix with suffix.

### PR DESCRIPTION
**Pull request by mr-c**
For `cwltool` `tmpdir_prefix` is not just a path to a directory, it can also include a suffix after the last path separator that become the prefix of the generated temporary directory.

### CODE REVIEW

File: src/toil/cwl/cwltoil.py [hunk 1]
```diff
@@ -1625,8 +1855,8 @@ def main(args: Union[List[str]] = None, stdout: TextIO = sys.stdout) -> int:
 
     # if tmpdir_prefix is not the default value, set workDir if unset, and move
     # workdir and the job store under it
-    if options.tmpdir_prefix != 'tmp':
-        workdir = tempfile.mkdtemp(dir=options.tmpdir_prefix)
+    if options.tmpdir_prefix != "tmp":
+        workdir = cwltool.utils.create_tmp_dir(options.tmpdir_prefix)
         os.rmdir(workdir)
```
Comment (mr-c) [inline src/toil/cwl/cwltoil.py:1860]: Actual change 1/2 (the rest is just from reformatting with `black`)

File: src/toil/cwl/cwltoil.py [hunk 2]
```diff
@@ -1637,14 +1867,15 @@ def main(args: Union[List[str]] = None, stdout: TextIO = sys.stdout) -> int:
             # --tmpdir-prefix.
             #
             # If set, workDir needs to exist, so we directly use the prefix
-            options.workDir = options.tmpdir_prefix
+            options.workDir = cwltool.utils.create_tmp_dir(options.tmpdir_prefix)
```
Comment (mr-c) [inline src/toil/cwl/cwltoil.py:1870]: Actual change 2/2 (the rest is just from reformatting with `black`)

**Review by DailyDreaming — APPROVED** [review]
This also seems to update the cwltool version and run CWL tests in parallel now according to the CPU count of the PC it's running on.  Approving, but please wait for tests to pass!

I haven't really given `Black` a chance before, but perhaps my disagreements with its decisions will be won over by the time spent not worrying about linting so much.  Since this is just in the CWL files, I'd say it's a good experiment to try.  I just wish we didn't have to smother the commit history.

File: src/toil/cwl/cwltoil.py [hunk 3]
```diff
@@ -1825,19 +2156,15 @@ def set_secondary(fileobj):
...
-        toilStageFiles(
-            toil,
-            outobj,
-            outdir,
-            destBucket=options.destBucket)
+        toilStageFiles(toil, outobj, outdir, destBucket=options.destBucket)
 
         if runtime_context.research_obj is not None:
-            runtime_context.research_obj.create_job(outobj, None, True)
+            runtime_context.research_obj.create_job(outobj, True)
```
Comment (mr-c) [inline src/toil/cwl/cwltoil.py:2162]: Another changed cwltool function call

**Comment by mr-c** [comment]
Thanks @DailyDreaming I didn't know that merge commits were turned off. I do recommend a single pass of `black` in its own PR. Active PRs can then do their own `black` formatting prior to merging in, to remove merge conflicts.

**Pull request merged**